### DISCUSSION TITLE
Overriding instance of `useId` used within `@react-aria` hooks

### DISCUSSION THREAD

**Discussion by ixTec** [adobe/react-spectrum#6374] [Q&A] [answered]
Is it possible to provide a custom `useId` functionality for the `@react-aria` hooks?

We use our own implementation of react's `useId`, so that it works for both react 18 and react < 18, but doesn't require a wrapper element (like `SSRProvider`). It also handles some additional logic for logging/analytics as well.

We'd much prefer it if we can also use this custom instance of `useId`, so were hoping if there was a way to override it for the hooks. Thanks!

**Comment by reidbarber** [comment, marked as answer]
No, but you could use something like [patch-package](https://github.com/ds300/patch-package) if you really need to override the behavior and are ok with the maintenance.

**Reply by snowystinger** [reply]
our copy of useId also works with all versions of React as well 
https://github.com/adobe/react-spectrum/blob/5cacbdbf88b8c6d7f5d3fb741c2b7ccddf8def7f/packages/%40react-aria/ssr/src/SSRProvider.tsx#L167

**Reply by ixTec** [reply]
`patch-package` was exactly what I needed, thank you!

Managed to get it working through some trial and error of rebuilding the `react-spectrum` repo locally 🙏
